apps/frontend/src/components/ui/ApplicationForm.tsx

```diff
@@ -58,24 +61,41 @@ const formSchema = z
       message: 'A szoba szám megadása kötelező, ha kolis vagy.',
     }
   );
-export default function ApplicationForm() {
+export default function ApplicationForm({ currentPeriod }: { currentPeriod: ApplicationPeriodEntity }) {
+  const user = useUser();
   const { toast } = useToast();
+  const effectCalledRef = useRef(false);
   const router = useRouter();
   const form = useForm<z.infer<typeof formSchema>>({
     resolver: zodResolver(formSchema),
     defaultValues: {
-      nickname: 'Bujdi Bohoc',
-      contact_email: 'email@gmail.com',
-      is_sch_resident: false,
-      room_number: 0,
+      nickName: 'Bujdi Bohoc',
+      email: 'email@gmail.com',
+      isSchResident: false,
+      roomNumber: 0,
       terms: false,
     },
   });
 
-  async function onSubmit(values: z.infer<typeof formSchema>) {
+  useEffect(() => {
+    if (user && !effectCalledRef.current) {
+      effectCalledRef.current = true;
+      form.setValue('nickName', user.data?.nickName || '');
+      form.setValue('email', user.data?.email || '');
+      form.setValue('isSchResident', user.data?.isSchResident || false);
+      form.setValue('roomNumber', user.data?.roomNumber || 0);
+      effectCalledRef.current = false;
+    }
+  }, [user.data]);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async function onSubmit({ terms, ...values }: z.infer<typeof formSchema>) {
     try {
-      const response = await axios.post('/api/application', values);
-      if (response.status === 200) {
+      const updateResponse = await api.patch('/users/me', values);
+      const response = await api.post('/application', {
```

Comment:
the user is able to send their application even if they fail to update their data?